src/ucp/rndv/rndv.c

```diff
@@ -287,12 +294,15 @@ void ucp_rndv_req_send_ack(ucp_request_t *ack_req, ucp_request_t *req,
     ucp_trace_req(req, "%s remote_req_id 0x%"PRIxPTR, ack_str, remote_req_id);
     UCS_PROFILE_REQUEST_EVENT(req, ack_str, 0);
 
-    ack_req->send.lane                = ucp_ep_get_am_lane(ack_req->send.ep);
-    ack_req->send.uct.func            = ucp_proto_progress_am_single;
-    ack_req->send.proto.am_id         = am_id;
-    ack_req->send.proto.status        = status;
-    ack_req->send.proto.remote_req_id = remote_req_id;
-    ack_req->send.proto.comp_cb       = ucp_request_put;
+    ack_req->send.lane                 = ucp_ep_get_am_lane(ack_req->send.ep);
+    ack_req->send.uct.func             = ucp_proto_progress_am_single;
+    ack_req->send.proto.am_id          = am_id;
+    ack_req->send.proto.status         = status;
+    ack_req->send.proto.remote_req_id  = remote_req_id;
+    ack_req->send.proto.comp_cb        = ucp_request_put;
+    /* ack_req->send.state.uct_comp is used when ack operation is failed */
+    ack_req->send.state.uct_comp.func  = ucp_rndv_ack_comp_cb;
```

Comment:
btw, does it mean that UCP receive (in case of RNDV/GET) operation will be uncompleted
I think we should complete UCP receive request in `ucp_rndv_ack_comp_cb()`
also, it probably, won't complete Fragmented RNDV operations

Reply:
it is completed before sending ack

Reply:
> it is completed before sending ack

yes, we discussed it with @hoopoepg 
missed this